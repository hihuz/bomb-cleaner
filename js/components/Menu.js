import React from 'react';

const Menu = ({ openModeDialog, openHSDialog, openAboutDialog }) => (
  <div className="menu">
    <button className="menu-button" onClick={openModeDialog}>Mode</button>
    <button className="menu-button" onClick={openHSDialog}>Best</button>
    <button className="menu-button" onClick={openAboutDialog}>About</button>
  </div>
);

export default Menu;
