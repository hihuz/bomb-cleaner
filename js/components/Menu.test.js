import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Menu from './Menu';

function setupButtonTest(n) {
  const openModeFn = jest.fn();
  const openHSFn = jest.fn();
  const openAboutFn = jest.fn();
  const component = shallow(<Menu
    openModeDialog={openModeFn}
    openHSDialog={openHSFn}
    openAboutDialog={openAboutFn}
  />);
  const button = component.find('.menu').children().at(n);
  button.simulate('click');
  return {
    openModeFn,
    openHSFn,
    openAboutFn
  };
}

test('Snapshot', () => {
  const component = shallow(<Menu />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Should call the openModeDialog prop when the first menu button is clicked', () => {
  const testFns = setupButtonTest(0);
  expect(testFns.openModeFn).toHaveBeenCalledTimes(1);
  expect(testFns.openHSFn).toHaveBeenCalledTimes(0);
  expect(testFns.openAboutFn).toHaveBeenCalledTimes(0);
});

test('Should call the openHSDialog prop when the second menu button is clicked', () => {
  const testFns = setupButtonTest(1);
  expect(testFns.openModeFn).toHaveBeenCalledTimes(0);
  expect(testFns.openHSFn).toHaveBeenCalledTimes(1);
  expect(testFns.openAboutFn).toHaveBeenCalledTimes(0);
});

test('Should call the openAboutDialog prop when the third menu button is clicked', () => {
  const testFns = setupButtonTest(2);
  expect(testFns.openModeFn).toHaveBeenCalledTimes(0);
  expect(testFns.openHSFn).toHaveBeenCalledTimes(0);
  expect(testFns.openAboutFn).toHaveBeenCalledTimes(1);
});

