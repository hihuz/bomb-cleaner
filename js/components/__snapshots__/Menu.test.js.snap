exports[`test Snapshot 1`] = `
<div
  className="menu">
  <button
    className="menu-button">
    Mode
  </button>
  <button
    className="menu-button">
    Best
  </button>
  <button
    className="menu-button">
    About
  </button>
</div>
`;
